src/azure.ts

```diff
@@ -101,3 +102,102 @@ export async function setSubscriptionAsync(context: Context, subscription: strin
         return { succeeded: false, result: null, error: [sr.stderr] };
     }
 }
+
+export async function configureCluster(context: Context, clusterType: string, clusterName: string, clusterGroup: string) : Promise<StageData> {
+    const downloadCliPromise = downloadCli(context, clusterType);
+    const getCredentialsPromise = getCredentials(context, clusterType, clusterName, clusterGroup, 5);
+
+    const [cliResult, credsResult] = await Promise.all([downloadCliPromise, getCredentialsPromise]);
+
+    const result = {
+        gotCli: cliResult.succeeded,
+        cliInstallFile: cliResult.installFile,
+        cliOnDefaultPath: cliResult.onDefaultPath,
+        cliError: cliResult.error,
+        gotCredentials: credsResult.succeeded,
+        credentialsError: credsResult.error
+    };
+    
+    return {
+        actionDescription: 'configuring Kubernetes',
+        result: { succeeded: cliResult.succeeded && credsResult.succeeded, result: result, error: [] }  // TODO: this ends up not fitting our structure very well - fix?
+    };
+}
+
+async function downloadCli(context: Context, clusterType: string) : Promise<any> {
+    const cliInfo = installCliInfo(context, clusterType);
+
+    const sr = await context.shell.exec(cliInfo.commandLine);
+    if (sr.code === 0) {
+        return {
+            succeeded: true,
+            installFile: cliInfo.installFile,
+            onDefaultPath: !context.shell.isWindows()
+        };
+    } else {
+        return {
+            succeeded: false,
+            error: sr.stderr
+        };
+    }
+}
+
+async function getCredentials(context: Context, clusterType: string, clusterName: string, clusterGroup: string, maxAttempts: number) : Promise<any> {
+    let attempts = 0;
+    while (true) {
+        attempts++;
+        const cmd = `az ${getClusterCommandAndSubcommand(clusterType)} get-credentials -n ${clusterName} -g ${clusterGroup}`;
+        const sr = await context.shell.exec(cmd);
+
+        if (sr.code === 0 && !sr.stderr) {
+            return {
+                succeeded: true
+            };
+        } else if (attempts < maxAttempts) {
+            await sleep(15000);
+        } else {
+            return {
+                succeeded: false,
+                error: sr.stderr
+            };
+        }
+    }
+}
+
+function installCliInfo(context: Context, clusterType: string) {
+    const cmdCore = `az ${getClusterCommandAndSubcommand(clusterType)} install-cli`;
```

Comment:
Shouldn't we check if cli already exists? I imagine users will many clusters and we don't need to re-download cli

Reply:
This is worth doing, but is a bit more invasive than just a rename.  Let's tackle it as a separate issue, rather than mixing it up in this PR.